scripts/west_commands/runners/intel_adsp.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for flashing with the Intel ADSP CAVS boards.'''
+
+import os
+import subprocess
+import re
+import signal
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+DEFAULT_CAVSTOOL= 'soc/xtensa/intel_adsp/tools/cavstool.py'
+
+class IntelAdspBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for the intel ADSP boards.'''
+
+    def __init__(self,
+                 cfg,
+                 remote_host,
+                 timeout,
+                 ):
+        super().__init__(cfg)
+
+        self.remote_host = remote_host
+        self.timeout = timeout
+        self.bin_fw = os.path.join(cfg.build_dir, 'zephyr', 'zephyr.ri')
+
+        self.zephyr_base = os.environ.get('ZEPHYR_BASE')
+        self.cavstool = os.path.join(self.zephyr_base, DEFAULT_CAVSTOOL)
+        self.build_dir = cfg.build_dir
+        self.board_dir = cfg.board_dir
+        self.platform = os.path.basename(self.board_dir)
+
+    @classmethod
+    def name(cls):
+        return 'intel_adsp'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        # Optional:
+        parser.add_argument('--remote-host',
+                            help='hostname of the remote targeting ADSP board')
+        parser.add_argument('-t', '--timeout', type=int, default=60,
+                            help='the pty where stdout goes to')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return IntelAdspBinaryRunner(cfg,
+                                    remote_host=args.remote_host,
+                                    timeout=args.timeout,
+                                    )
+
+    def do_run(self, command, **kwargs):
+        self.logger.info('Starting Intel ADSP runner')
+
+        if command == 'flash':
+            if re.search("intel_adsp_cavs", self.platform):
+                self.require(self.cavstool)
+                self.flash(**kwargs)
+            else:
+                self.run_sim(**kwargs)
+        else:
+            self.logger.info('Only support flash cmd now. Do nothing...')
```

Comment:
The `if` / `else` blocks are not necessary. The parent class will fail in `run()` before calling this `do_run()` method if the command is not supported in the `capabilities()` return value.

```python
        caps = self.capabilities()
        if command not in caps.commands:
            raise ValueError('runner {} does not implement command {}'.format(
                self.name(), command))
        self.do_run(command, **kwargs)
```

Reply:
Good to know, thanks! Remove it.